mopidy/core/library.py

```diff
@@ -242,6 +248,17 @@ def search(self, query=None, uris=None, exact=False, **kwargs):
             The ``exact`` keyword argument, which replaces :meth:`find_exact`.
         """
         query = _normalize_query(query or kwargs)
+
+        if kwargs:
+            warnings.warn(
+                'library.search() with keyword argument query is deprecated',
+                DeprecationWarning)
+
+        if not query:
+            warnings.warn(
+                'library.search() with an empty "query" argument deprecated',
```

Comment:
OK, I must have missed it because I expected it to be together with the `.. versionadded::` section or something.